prow/plugins/jira-linker/jira-linker.go

```diff
@@ -0,0 +1,106 @@
+package jira_linker
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName  = "jira-linker"
+	jiraPrefix  = "jira/"
+	noJiraLabel = jiraPrefix + "no-ticket"
+)
+
+var (
+	jiraRegex = regexp.MustCompile("([A-Z]+)-\\d+")
+)
+
+type githubClient interface {
+	AddLabel(owner, repo string, number int, label string) error
+	CreateComment(owner, repo string, number int, comment string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []string) (*pluginhelp.PluginHelp, error) {
+	// The Config field is omitted because this plugin is not configurable.
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The jira-linker plugin tries to detect JIRA references in PR titles and automatically adds a link to the ticket and a label",
+	}
+	return pluginHelp, nil
+}
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, pullRequestHandler, helpProvider)
+}
+
+func pullRequestHandler(pc plugins.PluginClient, event github.PullRequestEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, pc.PluginConfig.JiraLinker, &event)
+}
+
+func handle(gc githubClient, log *logrus.Entry, config plugins.JiraLinker, event *github.PullRequestEvent) error {
+	org := event.Repo.Owner.Login
+	repo := event.Repo.Name
+
+	labels, err := gc.GetIssueLabels(org, repo, event.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repo, event.Number)
+	}
+
+	matches := jiraRegex.FindStringSubmatch(event.PullRequest.Title)
+	if len(matches) > 1 {
+		ticketName := matches[0]
+		jiraTeamName := matches[1]
+
+		hasLabel := false
+		for _, candidate := range labels {
+			if candidate.Name == noJiraLabel {
+				gc.RemoveLabel(org, repo, event.Number, noJiraLabel)
```

Comment:
Is it possible for a PR to mention two tickets?

`FOO-123 fixes problem introduced by BAR-123`?

Reply:
I would expect, in that case, that the first match would be the one we'd care most about.  Though we could in that case extract both, I suppose.

Reply:
I hadn't considered this - maybe worth adding in future if it's a desired use-case? Not sure if it's worth adding to this PR as I'd prefer to distinguish based on the first ticket that appears so it'd probably be a non-negligible bit of work to add